hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1888,19 +1859,19 @@ public ExpiredOpenKeys getExpiredOpenKeys(Duration expireThreshold,
           } else {
             // add hsync'ed keys
             final KeyArgs.Builder keyArgs = KeyArgs.newBuilder()
-                .setVolumeName(info.getVolumeName())
-                .setBucketName(info.getBucketName())
-                .setKeyName(info.getKeyName())
-                .setDataSize(info.getDataSize());
-            java.util.Optional.ofNullable(info.getLatestVersionLocations())
+                .setVolumeName(openKeyInfo.getVolumeName())
+                .setBucketName(openKeyInfo.getBucketName())
+                .setKeyName(openKeyInfo.getKeyName())
+                .setDataSize(openKeyInfo.getDataSize());
```

Comment:
info used to have last commited datasize, commited blocks, but openKeyInfo may not have that information correctly. So IMO, can not use openKeyInfo for commit purpose.

Reply:
Good catch. Indeed, since it only updates `OpenKeyTable` on the first hsync call it won't always have the latest size:

https://github.com/apache/ozone/blob/HDDS-7593/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java#L272

**If** accurate data size here is a must, either `OpenKeyTable` have to be updated on every single hsync (which I don't like), or `getExpiredOpenKeys` have to resort to `KeyTable` again.

Reply:
Here in some case, keyInfo should be used, in other case, openKeyInfo should be used to commit the key. I did this check in OMRecoverLeaseRequest#doWork too.

Reply:
@sumitagrawl @ChenSammi Got it. Will use `KeyTable` in this case.

Reply:
done in 9ac270e4ffab432014fbf9b19efe7d8fd21646f1